hal/src/sercom/spi/async_api/dma.rs

```diff
@@ -83,43 +83,36 @@ where
     }
 }
 
-impl<C, S, R> SpiFuture<C, Rx, R, NoneT>
+impl<C, D, S, R, T> SpiFuture<C, D, R, T>
 where
     C: ValidConfig<Sercom = S>,
+    D: Capability,
```

Comment:
Here, instead of relaxing the capability, I would restrict it only to `Rx` (and not `RxDuplex`), and restrict the operating mode to `Slave`.

The `D: Receive` bound can probably be relaxed to `D: Capability` for the `take_rx_channel` method (line 61).